### ISSUE TITLE
TIME token does not display HH:MM:SS all the time.

### ISSUE THREAD

**Issue by ZaxonXP** [trizen/youtube-viewer#280]
If the videos on the result list are shorter then 1 hour then the display goes like `MM:SS`. However if the video is longer it displays as `HH:MM:SS`.

This breaks my custom layout as I put the *TIME* token at front. Can this be fixed in the next release?

Regards,
Piotr

**Comment by trizen** [comment]
Please check the extended version of `custom_layout_format`, which allows padding, alignment, colors and fixed-width formatting.

Example:

```perl
  custom_layout         => 1,
  custom_layout_format  => [
        {width => 3, text => "*NO*.", color => 'bold'},
        {width => '60%', text => "*TITLE*"},
        {width => '20%', text => "*AUTHOR*", color => 'yellow', align => 'right'},
        {width => 8, text => "*TIME*", color => 'blue', align => 'right'},
  ],
```

**Comment by ZaxonXP** [comment]
Tried to use the syntax from your example like this:
```
  custom_layout_format  => [
                            {width =>  8,       text => "*TIME*", align => 'right'},
                            {width => 11,       text => "*ID*"},
                            {width => 11,       text => "*PUBLISHED*"},
                            {width => '80%',   text => "*TITLE*\\n"},
                           ],
```
but the output I got from the `youtube-viewer --custom-layout "long run"` was empty. Then I pressed Enter to see the next page of results and I was the following output:

```
ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)ARRAY(0x560a83b9d830)

=>> Select one or more videos to play (:h for help)
>
```
It looks like there is something wrong.

**Comment by trizen** [comment]
Make sure to have latest master branch installed, as this feature was just added in https://github.com/trizen/youtube-viewer/commit/c3074371a9aa17ee840dddad12c147acbc144bf4.

**Comment by ZaxonXP** [comment]
OK. I was using script for updating the `youtube-viewer` which checks version on the github page:
```
lynx -dump github.com/trizen/youtube-viewer | grep "Version" | perl -ne 's/.*Version v*(\d+\.\d+\.\d+).*/$1/; print;' | sort | uniq | tail -1)
```
Since that version did not change, then I did not update the `youtube-viewer`. Maybe it is wise to update the version number each time there is change in master branch?

Now when I forced the update everything works as you described. :)

Many thanks!!!
